src/locales/__tests__/index.test.ts

```diff
@@ -31,71 +32,100 @@ describe('Locales', () => {
     jest.clearAllMocks();
   });
 
-  // if don't has the locale in AsyncStorage, should set default value ptBR
-  test('caso não possua o locale no AsyncStorage, deve setar o default ptBR', async () => {
+  // if don't has the locale in AsyncStorage, should set default value from 'react-native-loalize'
+  test('caso não possua o locale no AsyncStorage, deve setar o default de react-native-loalize', async () => {
     spyGetLocale.mockResolvedValue(null);
+    const spyFindBestAvailableLanguage = jest
+      .spyOn(RNLocalize, 'findBestAvailableLanguage')
+      .mockImplementation(() => ({
+        languageTag: 'en-US',
+        isRTL: false,
+      }));
 
     await initLocale();
 
     expect(spyGetLocale).toHaveBeenCalled();
-    expect(i18n.locale).toEqual('ptBR');
+    expect(spyFindBestAvailableLanguage).toHaveBeenCalledWith(
+      getAvailableLocales(),
+    );
+    expect(i18n.locale).toEqual('en-US');
+
+    spyFindBestAvailableLanguage.mockRestore();
+  });
+
+  // if don't has the locale in AsyncStorage, and system locale is not supported, should set it to 'pt-BR'
+  test('caso não possua o locale no AsyncStorage, localidade do sistema não é suportado, deve configurá-lo como pt-BR', async () => {
```

Comment:
Here you can write something like this:
"caso não possua o locale no AsyncStorage e o locale não é suportado pelo sistema, deve configurá-lo como pt-BR"